Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -856,90 +953,137 @@ namespace Terrain
-                    if (sector->m_hasData)
+                    AZStd::vector<HeightNormalVertex> meshHeightsNormals;
+
                     {
-                        UpdateSectorBuffers(*sector, meshHeightsNormals);
+                        SectorDataRequest request;
+                        request.m_samplesX = GridVerts1D;
+                        request.m_samplesY = GridVerts1D;
+                        request.m_worldStartPosition = sector->m_worldCoord.ToVector2() * gridMeters;
+                        request.m_vertexSpacing = gridMeters / GridSize;
+                        request.m_useVertexOrderRemap = true;
+
+                        GatherMeshData(request, meshHeightsNormals, sector->m_aabb, sector->m_hasData);
+                        if (sector->m_hasData)
+                        {
+                            UpdateSectorBuffers(*sector, meshHeightsNormals);
+                        }
+
+                        // Create AABBs for each quadrant for cases where this LOD needs to fill in a gap in a lower LOD.
+                        if (sector->m_aabb.IsValid())
+                        {
+                            float minX = sector->m_aabb.GetMin().GetX();
+                            float minY = sector->m_aabb.GetMin().GetY();
+                            float minZ = sector->m_aabb.GetMin().GetZ();
+                            float centerX = sector->m_aabb.GetCenter().GetX();
+                            float centerY = sector->m_aabb.GetCenter().GetY();
+                            float maxX = sector->m_aabb.GetMax().GetX();
+                            float maxY = sector->m_aabb.GetMax().GetY();
+                            float maxZ = sector->m_aabb.GetMax().GetZ();
+
+                            sector->m_quadrantAabbs.at(0) = AZ::Aabb::CreateFromMinMax(
+                                sector->m_aabb.GetMin(),
+                                AZ::Vector3(centerX, centerY, maxZ)
+                            );
+                            sector->m_quadrantAabbs.at(1) = AZ::Aabb::CreateFromMinMax(
+                                AZ::Vector3(centerX, minY, minZ),
+                                AZ::Vector3(maxX, centerY, maxZ)
+                            );
+                            sector->m_quadrantAabbs.at(2) = AZ::Aabb::CreateFromMinMax(
+                                AZ::Vector3(minX, centerY, minZ),
+                                AZ::Vector3(centerX, maxY, maxZ)
+                            );
+                            sector->m_quadrantAabbs.at(3) = AZ::Aabb::CreateFromMinMax(
+                                AZ::Vector3(centerX, centerY, minZ),
+                                sector->m_aabb.GetMax()
+                            );
+                        }
+                        else
+                        {
+                            AZStd::fill(sector->m_quadrantAabbs.begin(), sector->m_quadrantAabbs.end(), AZ::Aabb::CreateNull());
+                        }
```

Comment:
Suggestions:
1. Consider pulling into a helper method. It's extremely self-contained logic, and _might_ come in handy somewhere else at some point.
2. This could possibly be simplified a tiny bit by creating an AABB that represents the first quadrant (or even just the _size_ of the first quadrant), then using AABB::GetTranslated(offset) to move it to the other three quadrants.

Reply:
agreed on both counts.